### PR TITLE
Improve addon contract wiring

### PR DESCRIPTION
**Pull request by graupnerdemos**
@AlexMenschoid you know what to do ❤️ 🙈 🙉 🙊 

### PR Checklist

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [ ] Link all relevant tickets
- [ ] Move the tickets on the board accordingly

### CODE REVIEW

File: demosplan/DemosPlanCoreBundle/Addon/Registrator.php [hunk 1]
```diff
@@ -66,6 +65,8 @@ private function refreshAddonsYaml(): void
             'addons' => $this->addons,
         ];
 
+        // TODO: add a "this is autogenerated" note to the file
```
Comment (AlexMenschoid) [inline demosplan/DemosPlanCoreBundle/Addon/Registrator.php:69]: I did a quick search, but I have no idea how to include that in the dump. We might have to create the file first with the comment and then append the dump, which seems a bit too much trouble for a comment, I think.

File: demosplan/DemosPlanCoreBundle/Addon/RpcAddonAssetsLoader.php [hunk 2]
```diff
@@ -78,9 +78,10 @@ public function isTransactional(): bool
 
     public function validateRpcRequest(object $rpcRequest): void
     {
-        if (!$this->permissions->hasPermission('area_admin_consultations')) {
-            throw AccessDeniedException::missingPermission('area_admin_consultations');
-        }
+        // TODO: This needs a better validation
+        //if (!$this->permissions->hasPermission('area_admin_consultations')) {
+        //    throw AccessDeniedException::missingPermission('area_admin_consultations');
+        //}
```
Comment (AlexMenschoid) [inline demosplan/DemosPlanCoreBundle/Addon/RpcAddonAssetsLoader.php:84]: We should fix this before we merge it, I'm afraid

File: demosplan/DemosPlanCoreBundle/Resources/config/services.yml [hunk 3]
```diff
@@ -31,6 +31,193 @@ services:
...
+
+#    DemosEurope\DemosplanAddon\Contracts\Entities\EntityInterface:
+#        class:
+
+    DemosEurope\DemosplanAddon\Logic\Rpc\RpcErrorGeneratorInterface:
+        class: demosplan\DemosPlanCoreBundle\Logic\Rpc\RpcErrorGenerator
+
+    DemosEurope\DemosplanAddon\Permission\Validation\PermissionFilterValidatorInterface:
+        class: demosplan\DemosPlanCoreBundle\Permissions\PermissionResolver
+
```
Comment (AlexMenschoid) [inline demosplan/DemosPlanCoreBundle/Resources/config/services.yml:220]: Is this really necessary? And if so, shouldn't it be in the services.yml of the addon-repo?

**Review by AlexMenschoid — CHANGES_REQUESTED** [review]
I already added a couple of commits for little changes and a couple of things that were already solved but now broken again. I imagine, something went a bit wrong solving merge conflicts.

File: demosplan/DemosPlanCoreBundle/Addon/Registrator.php (hunk 1 above)
Comment (graupnerdemos) [reply demosplan/DemosPlanCoreBundle/Addon/Registrator.php:69]: It's not that complicated, I just forgot about it. We just dump into a string and prepend the comment, then write the string to the file.

File: demosplan/DemosPlanCoreBundle/Addon/RpcAddonAssetsLoader.php (hunk 2 above)
Comment (graupnerdemos) [reply demosplan/DemosPlanCoreBundle/Addon/RpcAddonAssetsLoader.php:84]: That's not my Baustellenflatterband-comment. 😇  (I'll have a look.)

File: demosplan/DemosPlanCoreBundle/Resources/config/services.yml (hunk 3 above)
Comment (graupnerdemos) [reply demosplan/DemosPlanCoreBundle/Resources/config/services.yml:220]: If you have a brilliant idea on how to bind services in the addon repo which the addon repo doesn't and mustn't know about, I'm all ears.

**Review by AlexMenschoid — APPROVED** [review]
looks as good as it can be to me. Let's go

**Pull request merged**